src/neuroconv/utils/json_schema.py

```diff
@@ -130,6 +130,9 @@ def get_schema_from_method_signature(method: Callable, exclude: list = None) ->
                     input_schema["properties"].update({param_name: dict(format="directory")})
             else:
                 arg = param.annotation
+                # in case __future__.annotations is imported, the annotation is a string
+                if isinstance(arg, str):
+                    arg = type(arg)
```

Comment:
Well, apparently all the new types under this behavior become `str` and so everything gets mapped to `string` in the JSON type map...

Reply:
Let's try it!

Reply:
Despite the input argument `eval_str` saying it handles exactly this case, it doesn't look like it actually does what is says anywhere, and ends up at https://github.com/shawwn/get-annotations/blob/main/get_annotations/__init__.py#L86 regardless